.github/workflows/pr-integration-tests.yml

```diff
@@ -0,0 +1,31 @@
+name: pull request integration tests
+
+on: pull_request
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        jdk: [1.8]
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up JDK
+      uses: actions/setup-java@v1
+      with:
+        java-version: ${{ matrix.jdk }}
+    - name: Cache local Maven repository
+      uses: actions/cache@v2
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-jdk${{ matrix.jdk }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-jdk${{ matrix.jdk }}-maven-
+    - name: Build
+      run: mvn -B clean install -Pquick,\!formatting
+    - name: Run tests
+      run: mvn -B integration-test -Pintegration -Dskip.surefire.tests
```

Comment:
Just as an FYI: never execute the 'integration-test' goal directly like this, because doing so makes maven skip the post-integration test phase (which is where the cleanup happens, e.g.  spinning down jetty servers and other resources). Always use `mvn verify` to execute integration testing.